src/openms/source/ANALYSIS/MAPMATCHING/QTClusterFinder.cpp

```diff
@@ -99,6 +100,132 @@ namespace OpenMS
+      medians_diffs.resize(ided_feat_rts.size());
+      Size c = 0;
+      for (auto& id_rts : ided_feat_rts)
+      {
+        #ifdef DEBUG_QTCLUSTERFINDER
+        std::cout << "Stats for " << id_rts.first << ": ";
+        for (const auto& rt : id_rts.second)
+        {
+          std::cout << rt << ", ";
+        }
+        std::cout << std::endl;
+        #endif
+        auto& rts = id_rts.second;
+        std::sort(rts.begin(),rts.end());
+        medians_diffs[c].first = rts[rts.size()/2];
+        medians_diffs[c].second.reserve(rts.size()-1);
+        Size i = 0;
+        for (const auto& rt : rts)
+        {
+          if (i == rts.size()/2) continue;
+          //TODO would relative diffs solve the RT dependency issue sufficiently? Probably not with non-linear shifts
+          medians_diffs[c].second.push_back(std::fabs(rt-medians_diffs[c].first));
+          i++;
+        }
+        c++;
+      }
+      //TODO we could also bin by equal sized RT bins, or by a fixed RT size
+
+      //TODO could be dependent on nr. of maps (e.g. with 5 maps you get [at least?] 4 diffs per ID already)
+      Size min_nr_diffs_per_bin = 50;
+
+      //TODO check if we can assume sorted
+      std::sort(medians_diffs.begin(), medians_diffs.end());
+
+      Size cnt = 0;
+      vector<double> tmp_diffs;
+      double start_rt = 0.;
+      double iqr_mult = 1.5;
+      double min_tolerance = 20;
+      double tol, iqr, q1, q3 = 0.;
+      OPENMS_LOG_DEBUG << "Calculating RT linking tolerance bins..." << std::endl;
+      for (const auto& med_diffs : medians_diffs)
+      {
+        if (tmp_diffs.size() > min_nr_diffs_per_bin)
+        {
+          std::sort(tmp_diffs.begin(), tmp_diffs.end());
+          // calculate allowed tolerance
+          //TODO find better measure. I think IQR only works for symmetric
+          q1 = quantile_(tmp_diffs, 0.25);
+          q3 = quantile_(tmp_diffs, 0.75);
+          iqr = q3 - q1;
+          tol = max(min_tolerance, max(fabs(q3 + iqr_mult * iqr), fabs(q1 - iqr_mult * iqr)));
+          bin_tolerances_.insert(make_pair(start_rt, tol));
+
+          OPENMS_LOG_DEBUG_NOFILE << "(RT: " << start_rt << ", Tol: " << tol << ")" << std::endl;
+          #ifdef DEBUG_QTCLUSTERFINDER
+          OPENMS_LOG_DEBUG_NOFILE << "Differences used: ";
+          for (const auto& diff : tmp_diffs)
+          {
+            OPENMS_LOG_DEBUG_NOFILE << diff << ", ";
+          }
+          OPENMS_LOG_DEBUG_NOFILE << std::endl;
+          #endif
+          tmp_diffs.clear();
+          start_rt = (med_diffs.first + medians_diffs[cnt-1].first)/2;
+        }
+        else
+        {
+          tmp_diffs.insert(tmp_diffs.end(), med_diffs.second.begin(), med_diffs.second.end());
+        }
+        cnt++;
+      }
+      // calculate allowed tolerance
+      std::sort(tmp_diffs.begin(), tmp_diffs.end());
+      q1 = quantile_(tmp_diffs, 0.25);
+      q3 = quantile_(tmp_diffs, 0.75);
```

Comment:
we have MAD in statistics functions which might also be an option

Reply:
Yes, actually I thought about something completely different, since I think measures of spread are often used for symmetric distributions only. I currently use absolute differences.
But I am not sure. It kind of works already.